kolibri/core/fields.py

```diff
@@ -94,3 +97,23 @@ def get_db_prep_value(self, value, connection, prepared=False):
     def value_from_object_json_compatible(self, obj):
         if self.value_from_object(obj):
             return create_timezonestamp(self.value_from_object(obj))
+
+
+class JSONField(JSONFieldBase):
+    def from_db_value(self, value, expression, connection, context):
+        if isinstance(value, string_types):
+            try:
+                return json.loads(value)
+            except ValueError:
+                pass
+
+        return value
+
+    def to_python(self, value):
+        if isinstance(value, string_types):
+            try:
+                return json.loads(value)
+            except ValueError:
+                pass
+
+        return value
```

Comment:
Will it always be the correct behavior to swallow `ValueError` and fall back on returning the original value?

It's a little unclear to me why we'd call these functions on data that's already in the target format.